src/seqn/trans.lisp

```diff
@@ -0,0 +1,399 @@
+                                    (cadrobjnum (vamp:const cadrobj)))
+                                (cond ((= carobjnum cadrobjnum 0) (vamp:make-constant :const 0))
+                                      ((and (= carobjnum 0)
+                                            (not (= cadrobjnum 1)))
+                                       ifone)
+                                      ((and (= carobjnum 0)
+                                            (= cadrobjnum 1))
+                                       car)
+                                      ((and (not (= carobjnum 1))
+                                            (= cadrobjnum 0))
+                                       (infix-creation :*
+                                                       ifzero
+                                                       carobj))
+                                      ((and (= carobjnum 1)
+                                            (= cadrobjnum 0))
+                                       ifzero)
+                                      ((and (= carobjnum cadrobjnum 1))
+                                       (infix-creation :+
+                                                       ifzero
+                                                       car))
+                                      (t output))))
+                             ((typep carobj 'vamp:constant)
+                              (let ((carobjnum (vamp:const carobj)))
+                                (cond ((= carobjnum 0) (infix-creation :*
+                                                                       car
+                                                                       cadrobj))
+                                      ((= carobjnum 1) (infix-creation :+
+                                                                       ifzero
+                                                                       ifone))
+                                      (t output))))
+                             ((typep cadrobj 'vamp:constant)
+                              (let ((cadrobjnum (vamp:const cadrobj)))
+                                (cond ((= cadrobjnum 0) (infix-creation :*
+                                                                        ifzero
+                                                                        carobj))
+                                      ((= cadrobjnum 1) (infix-creation :+
+                                                                        (infix-creation :*
+                                                                                        ifzero
+                                                                                        carobj)
+                                                                        car))
+                                      (t  output))))
+                             (t output))))
+         (prod-list left
+                    right)))))
```

Comment:
This looks big, can you abstract it at all? I am seeing some repeat patterns, but can we somehow make this cleaner? I should be able to at a glance figure out some things, this is a bit intimidating